src/mpid/ch4/src/ch4_coll_globals_default.c.in

```diff
@@ -2,11 +2,10 @@
 #include <mpidimpl.h>
 #include "ch4_impl.h"
 
-/* container traverse function to be implemented  */
-/* e.g. return ((void *) ((char *)container) + sizeof(MPIDIG_coll_algo_generic_container_t)); */
+/* Traverses the container list to help serialize the collective containers. */
 const void *MPIDI_coll_get_next_container(const void *container)
 {
-    return NULL;
+    return ((char *) container) + sizeof(MPIDIG_coll_algo_generic_container_t);
```

Comment:
Is the type of the input array always the same? If so, it should be able to avoid this cast and just increment the array index, no?

Reply:
The type of input and output can be different, they can be for netmod, shm or composite algorithms. However, the MPIDIG_coll_algo_generic_container is a union type of all these three. 
I think it is ok to change this function to take MPIDIG container and returns MPIDIG_container. 
But now the caller of this function may need to cast the parameter from netmod/posix container to MPIDIG_ container type.

Reply:
OK, that helps my understanding.